src/backend/distributed/utils/maintenanced.c

```diff
@@ -345,6 +366,42 @@ CitusMaintenanceDaemonMain(Datum main_arg)
 
 
 /*
+ * MetadataTableAccessible returns whether the given metadata table exists
+ * and can be locked using an AccessShareLock.
+ */
+static bool
+MetadataTableAccessible(const char *tableName)
+{
+	bool tableAccessible = false;
+
+	Oid relationId = get_relname_relid(tableName, PG_CATALOG_NAMESPACE);
```

Comment:
> What do you mean by trylock callback?

Instead of having RangeVarGetRelidExtended acquire the lock, acquire it in the callback. But there's plenty uglyness around it, so it's not really good.

> We then tried putting it in PG_TRY, but that turned out to be a bad idea in a background worker.

Hm. That's where you got the above error messages?